sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -93,8 +93,8 @@ object QueryCompilationErrors {
 
   def unsupportedIfNotExistsError(tableName: String): Throwable = {
```

Comment:
shall we add a `QueryCompilationErrorsSuite` and test this error?

Reply:
`InsertIntoSQLOnlyTests` has a test for the error already. BTW, the test from `InsertIntoSQLOnlyTests` runs as part of:

- V1WriteFallbackSessionCatalogSuite
- DataSourceV2SQLSuite
- DataSourceV2SQLSessionCatalogSuite
- DataSourceV2DataFrameSuite
- DataSourceV2DataFrameSessionCatalogSuite

@cloud-fan How about to create the `QueryCompilationErrorsSuiteBase` trait/abstract class and include it to those test suites?

Reply:
I added the base test suites `QueryCompilationErrorsSuiteBase` + `SessionCatalogTestBase` and new test suites for compilation errors:

- QueryCompilationErrorsDSv2Suite
- QueryCompilationErrorsDSv2SessionCatalogSuite
- QueryCompilationErrorsV1WriteFallbackSuite